Homework-8/script.js

```diff
@@ -0,0 +1,211 @@
+// Task 1
+
+// const makeCounter = () => {
+//   let count = 0;
+
+//   return () => {
+//     return count++;
+//   };
+// };
+
+// // independent counters
+
+// const counter = makeCounter();
+// const counter2 = makeCounter();
+
+// console.log(counter()); // 0
+// console.log(counter()); // 1
+
+// console.log(counter2()); // 0
+// console.log(counter2()); // 1
+
+// Task 2
+
+// function Counter() {
+//   let count = 0;
+
+//   this.up = function() {
+//     return ++count;
+//   };
+//   this.down = function() {
+//     return --count;
+//   };
+// }
+
+// const counter = new Counter();
+
+// console.log(counter.up()); // 1
+// console.log(counter.up()); // 2
+// console.log(counter.down()); // 1
+
+// Task 3
+
+// const phrase = "Hello";
+
+// if (true) {
+//   const user = "John";
+
+//   function sayHi() {
+//     console.log(`${phrase}, ${user}`);
+//   }
+// }
+
+// sayHi(); // Hello, John
+
+// Task 4
+
+// const sum = num1 => {
+//   return (num2) => {
+//       return num1 + num2;
+//   }
+// };
+
+// console.log(sum(2)(2));
+// console.log(sum(-5)(20));
+
+// Task 5
+
+// const arr = [1, 2, 3, 4, 5, 6, 7];
+
+// const inBetween = (from, to) => {
+//   return function() {
+//     if (arguments[0] >= from && arguments[0] <= to) {
+//       console.log(arguments[0]);
+//       return true;
+//     }
+//     return false;
+//   };
+// };
+
+// const inArray = arr => {
+//   return function() {
+//     return arr.includes(arguments[0]);
+//   };
+// };
+
+// console.log(arr.filter(inBetween(3, 6)));
+// console.log(arr.filter(inArray([1, 2, 10])));
```

Comment:
```
const arr = [1, 2, 3, 4, 5, 6, 7];

const inBetween = (from, to) => {
  return item => {
    return item >= from && item <= to ? true : false;
  };
};

const inArray = arr => {
  return item => {
    return arr.includes(item);
  };
};

console.log(arr.filter(inBetween(3, 6)));
console.log(arr.filter(inArray([1, 2, 10])));
```
1. First of all, keep it simple. 
2. Secondary, I'm not sure, that you correctly understood how it works.

arr.filter should receive callback function, and filter logic provides you as a first argument element of an array.